### PR TITLE
add command to output block0 initials

### PR DESCRIPTION
**Pull request by Zeegomo**

### CODE REVIEW

**Review by enzoc4 — APPROVED** [review]
LGTM

File: src/bin/cli/snapshot/mod.rs [hunk 1]
```diff
@@ -0,0 +1,60 @@
...
+    OutputFile(#[from] OutputFileError),
+    #[error(transparent)]
+    OutputFormat(#[from] OutputFormatError),
+}
+
+impl SnapshotCmd {
+    pub fn exec(self) -> Result<(), Error> {
+        let raw_snapshot: RawSnapshot = serde_json::from_reader(File::open(&self.snapshot)?)?;
+        let initials = Snapshot::from_raw_snapshot(raw_snapshot, self.threshold)
+            .to_block0_initials(self.discrimination);
```
Comment (dkijania) [inline src/bin/cli/snapshot/mod.rs:52]: What vitup expects is that root of json is initials, like below:

```
{
    "initial": [
        {
            "fund": [
                {
```

But this is not a big problem. However there is another discrepancy as funds collections are split by 100 items. Question if not splitting is ok? i remember that there was some upper limit of items in fund

Comment (Zeegomo) [reply src/bin/cli/snapshot/mod.rs:52]: I guess that vitup expects a `Block0Configuration`, while here the only thing provided is the list of initials, so I would keep it that way to make it explicit that this tools only provides that.

For the second issue, I think that has to do with limits in fragment size, i.e. it's not logically wrong to have all fragments in a single `Initial::Fund`, but only 255 will fit in a fragment on chain. Tbh, we do implement `impl TryFrom<Initial> for Fragment`, so there is some expectation that you should be able to convert between the two.

If it's not too much work on vitup, I would suggest we keep this tool as blockchain agnostic as possible, and change this to return just a `Vec<InitialUTxO>` (i.e. just a collection of inital funds), leaving splitting to deployment tools.

File: src/bin/cli/snapshot/mod.rs [hunk 2]
```diff
@@ -0,0 +1,60 @@
...
+        let mut out_writer = self.output.open()?;
+        let content = self
```
Comment (Mr-Leshiy) [inline src/bin/cli/snapshot/mod.rs:54]: maybe better to change places
```
let mut out_writer = self.output.open()?;
```
and 
```
let content = self.output_format
             .format_json(serde_json::to_value(initials)?)?;
``` 
So if some errors appears `format_json(serde_json::to_value(initials)?)` we will not even open output file

Comment (Zeegomo) [reply src/bin/cli/snapshot/mod.rs:54]: We would close it right after an error anyway and I don't expect `serde_json::to_value(initials)` to ever fail, so I don't think it makes much of a difference

File: src/bin/cli/snapshot/mod.rs (hunk 1 above)
Comment (dkijania) [reply src/bin/cli/snapshot/mod.rs:52]: Agree!

**Review by dkijania — APPROVED** [review]

**Pull request merged**